### ISSUE TITLE
Try changing the `lib` compiler option to es2015 or later

### ISSUE THREAD

**Issue by apiel** [vega/ts-json-schema-generator#103]
When I try to use ts-json-schema-generator with angular I get an error:

```
/node_modules/@angular/common/src/directives/ng_class.d.ts(48,34): error TS2583: Cannot find name 'Set'. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
/node_modules/@angular/common/src/pipes/keyvalue_pipe.d.ts(47,9): error TS2583: Cannot find name 'Map'. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
/node_modules/@angular/common/src/pipes/keyvalue_pipe.d.ts(50,9): error TS2583: Cannot find name 'Map'. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
...
```
This error is the same as https://github.com/vega/ts-json-schema-generator/issues/47

To reproduce:
```
npm install -g @angular/cli
ng new my-app
cd my-app
npm install ts-json-schema-generator
npx ts-json-schema-generator  --path src/main.ts --type string
```

In the other issue, you propose to use `ts-json-schema-generator --path tsconfig.json` but in my case I really need to specify the file to search for types. I don't want ts-json-schema-generator to scan the whole project.

Also, if I copy the node_module outside the project and install the dependency, it is then working. 
```
cp -r  my-app/node_modules/ts-json-schema-generator .
cd ts-json-schema-generator
npm install
npx ts-json-schema-generator  --path ../my-app/src/main.ts --type string
```
But I need ts-json-schema-generator to be part of the dependency of my project.

**Comment by apiel** [comment]
If I add `lib: ["es2015"],` to `createProgramFromGlob` in factory/program.ts the problem goes away but then I get another error.
```
function createProgramFromGlob(fileGlob: string): ts.Program {
    return ts.createProgram(glob.sync(path.resolve(fileGlob)), {
        noEmit: true,
        emitDecoratorMetadata: true,
        experimentalDecorators: true,
        target: ts.ScriptTarget.ES5,
        lib: ["es2015"],
        module: ts.ModuleKind.CommonJS,
        strictNullChecks: false,
    });
}
```
Might be because it need more info in the config of `createProgram`. But what I still don't understand, is why does it work when it is outside the project...

**Comment by apiel** [comment]
If I am using the tsconfig as path and hardcode the path of the file I want to seach for type it is then working:
`npx ts-json-schema-generator --path tsconfig.app.json --type string`

Then in node_modules/ts-json-schema-generator/dist/factory/program.js
```
function createProgramFromConfig(configFile) {
    const config = ts.parseConfigFileTextToJson(configFile, ts.sys.readFile(configFile));
    if (config.error) {
        throw new DiagnosticError_1.DiagnosticError([config.error]);
    }
    else if (!config.config) {
        throw new LogicError_1.LogicError(`Invalid parsed config file "${configFile}"`);
    }
    const parseResult = ts.parseJsonConfigFileContent(config.config, ts.sys, path.dirname(configFile), {}, configFile);
    parseResult.options.noEmit = true;
    delete parseResult.options.out;
    delete parseResult.options.outDir;
    delete parseResult.options.outFile;
    delete parseResult.options.declaration;
    return ts.createProgram(
        // parseResult.fileNames,
        ["/some_absolute_path/file.ts"]
        parseResult.options,
    );
}
```
So it would be good to be able to have more parameter available, 1 for the path of the file `--path 'my/project/**.*.ts'` and one for the the tsconfig `--tsconfig tsconfig.ts`. Also you could use a package like find-up to search for the tsconfig file.

**Comment by apiel** [comment]
I fixed it in my repo, have a look at https://github.com/apiel/isomor/commit/9d0f1c80b52e0857a2e79824cfa3339eaef13f6b

Unfortunatelly, I don't have time to update the test.

**Comment by domoritz** [comment]
Yeah, I think we should take both a path to a file and a tsconfig file. Can you send a PR to add this feature?

**Comment by apiel** [comment]
I will try tomorrow, if I have time but I don't think I will have time to update the unit test.

Also would be great to get merged the function types https://github.com/apiel/isomor/blob/master/packages/isomor-json-schema-generator/src/NodeParser/FuncTypeNodeParser.ts unfortunatelly you didn't answer my question in the other issue. But I am actually not sure about the approch I used, maybe i should have used tuple validation: https://json-schema.org/understanding-json-schema/reference/array.html#tuple-validation

**Comment by domoritz** [comment]
Which issue did I not comment on?

**Comment by apiel** [comment]
PR is here https://github.com/vega/ts-json-schema-generator/pull/109

**Comment by apiel** [comment]
Concerning the other issue, here is my comment: https://github.com/vega/ts-json-schema-generator/issues/98#issuecomment-496252968

**Comment by AndrewLeedham** [comment]
@apiel I updated your PR and it was merged as of `v0.51.0`. So this is now resolved 😃
